app/helpers/spree/admin/base_helper.rb

```diff
@@ -62,6 +62,17 @@ def preference_field_for(form, field, options)
         end
       end
 
+      def preference_field_for_text_field(form, field, options, object)
```

Comment:
This needs a comment briefly explaining what's happening and the interface required on the object.
Maybe something like:
```
# Here we show a text field for all string fields except when the field name ends in "_from_list", in that case we render a dropdown
# In this specific case, to render the dropdown correctly, the object provided must have a method named like "#{field}_values" that returns an array with the string options to be listed.
```